core/deployment/src/main/java/io/quarkus/deployment/dev/RuntimeUpdatesProcessor.java

```diff
@@ -585,6 +587,27 @@ public boolean doScan(boolean userInitiated, boolean forceRestart) {
         }
     }
 
+    /**
+     * This notifies registered extensions of "no-restart" changed files.
+     *
+     * @param noRestartChangedFiles the Set of changed files
+     */
+    public void notifyExtensions(Set<String> noRestartChangedFiles) {
+        if (lastStartIndex == null) {
+            // we don't notify extensions if the application never started
+            return;
+        }
+        try {
+            scanLock.lock();
```

Comment:
Shouldn't it be `lock` before the `try` like all other such cases?

Reply:
Done, I also added `codeGenLock`